compiler/qsc_parse/src/lex/cooked.rs

```diff
@@ -365,7 +366,11 @@ impl<'a> Lexer<'a> {
                 self.expect(Single::Amp, TokenKind::ClosedBinOp(op))?;
                 Ok(self.closed_bin_op(op))
             }
-            Single::Apos => Ok(TokenKind::Apos),
+            Single::Apos => {
+                self.tokens
+                    .next_if(|t| matches!(t.kind, raw::TokenKind::Ident));
+                Ok(TokenKind::AposIdent)
```

Comment:
This treats the identifier after the apostrophe as optional, meaning that an apostrophe followed by a non-ident will still be cooked into AposIdent. There is no other token that uses apostrophe, so it should be an error.